lib/aggregator.js

```diff
@@ -106,14 +107,27 @@ async function aggregateAndSendToCoreAsync() {
     try {
       let hash = crypto.createHmac('sha256', HMAC_KEY)
       let hmac = hash.update(env.NODE_TNT_ADDRESS).digest('hex')
-      let submitResult = await submitHashToCoreAsync(
-        aggregationMerkleRoot,
-        hmac
-      )
+
+      let submitResult
+      try {
+        submitResult = await submitHashToCoreAsync(aggregationMerkleRoot, hmac)
+
+        // Submission to Core was successful, purge hashes that were delivered from RocksDB
+        await rocksDB.deleteBatchAsync(hashDataForTreeDeleteOpts)
+      } catch (error) {
+        // There was an error submitting to core, go ahead and persist all of the hashes that
```

Comment:
@jasonbukowski @michael-iglesias Is there a better way to do this than delete on first read and write back on error?